### ISSUE TITLE
Add runtime hash validation of all code files

### ISSUE THREAD

**Issue by chris001** [salesagility/SuiteCRM#862]
It would be very good from an information security point of view, to add the feature of having the suite core code be either immutable or capable of detecting changes to the core code files, by means of runtime hash validation.  Obviously, each new release would come with a new set of hashes due to all of the changed code files.  This should be fine, no module or add on should need to change any core code file, since the core application code can be extensible with modules themes and connectors that are isolated from the core app by the classic API, and or a newer REST v10+ compatible API.

At the very minimum, the hash validation must notify the admin user when core files have been changed so that it could be looked into manually and verified that an information stealing shell or trojan has not been planted as the result of a security breach.

**Comment by samus-aran** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Add runtime hash validation of all code files](https://trello.com/c/sEMpJHRH/123-add-runtime-hash-validation-of-all-code-files)

**Comment by samus-aran** [comment]
We have now set up a new home for suggestions at Trello.  All github issues that were labeled 'suggestion' have been moved and will be closed.  Certain ones will be progressed within the new Suggestion Box and may be re-opened.

Announcement of moving Suggestions:
https://suitecrm.com/forum/suggestion-box/13691-moving-suggestions-from-github-to

New SuiteCRM Suggestion Box
https://trello.com/b/Ht7LbMqw/suitecrm-suggestion-box
